application-ownership-models/README.md

```diff
@@ -0,0 +1,116 @@
+- Giant Swarm team is packaging app as a Helm chart in the app repository, by building required Helm templates for deploying re-tagged container image and optionally using existing upstream Kubernetes manifests instead of writing Helm templates from scratch.
+- Giant Swarm is repackaging or importing existing upstream Helm charts into Giant Swarm application repositories.
+
+## 2. Sole or shared application ownership
+
+While different application ownership models based on application lifecycle stages are relatively clear and it's obvious which model is used for every application, for some apps is much less evident which team should own them, and if there can even be just one owning team, or multiple teams cannot be avoided due to shared nature of the app.
+
+Additionally, while application lifecycle stages split the app ownership between external (upstream) and internal (Giant Swarm) teams, sole or shared ownership is about how we handle app ownership internally within Giant Swarm.
+
+### 2.1. Sole application ownership
+
+The is the classical and usually the obvious case. A Giant Swarm team develops the application, in case of full application lifecycle ownership described in the previous section, or the team takes care of release engineering and operations for the app, in case of partial application lifecycle ownership described above.
+
+### 2.2. Shared application ownership
+
+And this is the tricky part, one of the main reasons for which this whole RFC is written.
+
+Shared application ownership applies in some cases of partial application lifecycle ownership, so in a nutshell, the question here is - which Giant Swarm team owns some upstream application? Can it even be one team, or multiple teams must share ownership? If multiple teams share ownership, how does that work in practice?
+
+There is no silver bullet for all upstream applications, so here we need clear and robust, but also flexible ownership model.
+
+Before trying to come up with some definition, let's first check few examples.
+
+#### 2.2.1. Examples of shared applications
+
+While there are more examples, let's take the following two that should paint the picture and describe the issue of shared ownership clearly enough.
+
+**Cluster API provider-independent controllers**
+
+There are three CAPI controllers that are equally used by KaaS teams (Rocket, Hydra, Clippy) - core, kubeadm control plane and kubeadm bootstrap controller. They are all packaged together in the cluster-api-app. All KaaS teams have equal stakes in this app and equal responsibilities.
+
+Which teams is responsible for maintaining the application repository, improving it, automating it, taking care of renovate and other automated pull requests? Which teams takes care of upgrading the app to the latest upstream? Every team for themselves when they have a need? This is not working very efficiently.
```

Comment:
This was one of the main drivers why we thought about KaaS restructuring, having a single Lifecyle mgmt team for k8s will make ownership clearer. [We have just produced the first proposal](https://github.com/giantswarm/giantswarm/issues/23396#issuecomment-1413962194). WDYT?

Reply:
I was following the restructuring proposal doc, left a few comments.

The above here are just few examples, definitely not the only ones. I will add few more.

While this RFC could solve some issues that we are also trying to solve with team restructuring, this RFC and the restructuring serve a different purpose and are not mutually exclusive.

The ownership of many components will change and/or become more clear with team restructuring, but that just one of the effects of the restructuring, not its goal.

If you take a look at our "Ownership of component" spreadsheet, there we have almost 300 components. While assigning all those components to teams, we did our best, but still many of those components got assigned to teams that never worked on them, or to teams that were already overwhelmed and weren't exactly happy to get more responsibility. We have cases of components shared across teams (e.g. Giant Swarm modules used across many projects), our legacy components that we still have to maintain, patch at least to fix CVEs that are coming our way from upstream dependencies, etc. In many of those cases those applications/components are not super nicely maintained because they are not exactly a team priority.

So there are cases where either multiple teams really need something, or where almost none of the teams wants/needs it, and then different manifestations of those problems in different applications/components. All these diverse cases are what this RFC is targeting, by defining different models of application/component ownership (basically adding a rotation model for shared ownership), so we can more easily assign applications/components to teams, by having an option to assign it to multiple teams if that is needed.